src/fsharp/FSharp.Compiler.LanguageServer/State.fs

```diff
@@ -2,11 +2,102 @@
 
 namespace FSharp.Compiler.LanguageServer
 
+open System
+open System.Collections.Concurrent
+open System.IO
+open FSharp.Compiler.SourceCodeServices
+open Microsoft.Build.Construction
+open Microsoft.Build.Definition
+open Microsoft.Build.Evaluation
+open StreamJsonRpc
+
 type State() =
 
+    let checker = FSharpChecker.Create()
+
+    let sourceFileToProjectMap = ConcurrentDictionary<string, FSharpProjectOptions>()
```

Comment:
Saw your comment on linked files. You'll also be worrying about multiple TFMs as well in the future.

Reply:
Added comments stating that we'll revisit those issues in the future.